Src/Support/Google.Apis/Requests/ClientServiceRequestExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System;
+
+namespace Google.Apis.Requests
+{
+    /// <summary>
+    /// Extension methods for request objects.
+    /// </summary>
+    public static class ClientServiceRequestExtensions
+    {
+        /// <summary>
+        /// Allows a request to be configured in a fluent manner, where normally separate statements are required
+        /// after the request creation method is called.
+        /// </summary>
+        /// <typeparam name="TRequest"></typeparam>
+        /// <param name="request"></param>
+        /// <param name="configure"></param>
+        /// <returns></returns>
+        public static TRequest WithOptions<TRequest>(this TRequest request, Action<TRequest> configure)
```

Comment:
What else can you do with a request? We don't have a way to enforce that only options are changed, so the name is just a hint, maybe.
Can we change the name maybe to WithConfiguration?

Reply:
Or `WithParameters`? `WithProperties`? `Configure` instead of `With` at all?

Happy to bikeshed this aspect for a bit.

Reply:
I think like plain `Configure` better. `WithAnything` might give the idea that a new instance needs to be returned?

Reply:
Done.